src/Interpreters/CrashLog.cpp

```diff
@@ -84,5 +84,6 @@ void collectCrashLog(Int32 signal, UInt64 thread_id, const String & query_id, co
 
         CrashLogElement element{static_cast<time_t>(time / 1000000000), time, signal, thread_id, query_id, trace, trace_full};
         crash_log_owned->add(element);
+        crash_log_owned->flush(true);
```

Comment:
This is not the best option, because a crash can happen in multiple threads in a short time.

Reply:
Missing a comment.